src/rgw/rgw_bucket.cc

```diff
@@ -295,29 +295,28 @@ int rgw_bucket_instance_remove_entry(RGWRados *store, string& entry, RGWObjVersi
   return store->meta_mgr->remove_entry(bucket_instance_meta_handler, entry, objv_tracker);
 }
 
+// parse instance in format: [tenant:]instance[:shard_id], returning
```

Comment:
actually it's not even safe to assume that all buckets will even have a bucket instance (e.g., buckets created pre dumpling). I'm not sure whether it's going to always have the same format (could be that the zone was added later). We really shouldn't rely on that string format.